AsyncDisplayKit/ASMapNode.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+#import <MapKit/MapKit.h>
+@interface ASMapNode : ASControlNode
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate NS_DESIGNATED_INITIALIZER;
+/**
+ This is the snapshot shot image node, this will be hidden (but not nil) when .liveMap = YES
+ */
+@property (nonatomic, readonly) ASImageNode *mapImage;
+/**
+ This is the ASDisplayNode that backs the MKMapView. This will be nil if .liveMap = NO. To access the underlying MKMapView, in order to set a delegate for example, use (MKMapView *)mapView.view;
+ */
+@property (nonatomic, readonly) ASDisplayNode *mapView;
+/**
+ Set this to YES to turn the snapshot into an interactive MKMapView and vice versa. Defaults to NO.
+ */
+@property (nonatomic, assign, getter=isLiveMap) BOOL liveMap;
+/**
+ @abstract Explicitly set the size of the map and therefore the size of ASMapNode. Defaults to CGSizeMake(constrainedSize.max.width, 256).
```

Comment:
@appleguy The reason it defaults to 256 high is because the MKMapSnapshotOptions used in MKMapSnapshotter defaults to 256x256 so I took this as a sensible default. I agree with you and @nguyenhuy that ASMapNode should have no fixed form/size so any solution that accomplishes this would be best, I prefer for ASMapNode to take the height that the parent node gives it, rather than to calculate a intrinsic size for ASMapNode